inc/functions.php

```diff
@@ -1807,6 +1807,20 @@ function is_moderator($fid=0, $action="", $uid=0)
 	}
 
 	$user_perms = user_permissions($uid);
+
+	$hook_args = array(
+		'fid' => $fid,
+		'action' => $action,
+		'uid' => $uid,
+		'user_perms' => &$user_perms',
```

Comment:
It is unnecessary to pass this over as you already passed `$uid` and `user_permissions()` handles caching.

Plus with the suggested change to usage of `$hook_args['is_moderator']` it is superfluous to pass as developers can already affect the returned value of `is_moderator()`.

Reply:
@lairdshaw do you have a use scenario where passing `$user_perms` is necessary considering my message above?

Reply:
I don't think so, no. It seems reasonable to exclude it, especially considering that later in the function there are similar variables which wouldn't be passed to the hook, e.g., `$forumpermissions` and `$modcache`.

Reply:
```suggestion
```